src/topology/maps.lean

```diff
@@ -13,130 +13,200 @@ local attribute [instance] classical.prop_decidable
 
 lemma embedding_open {f : α → β} {s : set α}
   (hf : embedding f) (h : is_open (range f)) (hs : is_open s) : is_open (f '' s) :=
-let ⟨t, ht, h_eq⟩ := by rw [hf.right] at hs; exact hs in
-have is_open (t ∩ range f), from is_open_inter ht h,
-h_eq ▸ by rwa [image_preimage_eq_inter_range]
+inducing_open hf.1 h hs
 
 lemma embedding_is_closed {f : α → β} {s : set α}
   (hf : embedding f) (h : is_closed (range f)) (hs : is_closed s) : is_closed (f '' s) :=
-let ⟨t, ht, h_eq⟩ := by rw [hf.right, is_closed_induced_iff] at hs; exact hs in
-have is_closed (t ∩ range f), from is_closed_inter ht h,
-h_eq.symm ▸ by rwa [image_preimage_eq_inter_range]
+inducing_is_closed hf.1 h hs
 
 lemma embedding.map_nhds_eq [topological_space α] [topological_space β] {f : α → β}
   (hf : embedding f) (a : α) (h : range f ∈ nhds (f a)) : (nhds a).map f = nhds (f a) :=
-by rw [hf.2]; exact map_nhds_induced_eq h
+inducing.map_nhds_eq hf.1 a h
 
 lemma embedding.tendsto_nhds_iff {ι : Type*}
   {f : ι → β} {g : β → γ} {a : filter ι} {b : β} (hg : embedding g) :
   tendsto f a (nhds b) ↔ tendsto (g ∘ f) a (nhds (g b)) :=
-by rw [tendsto, tendsto, hg.right, nhds_induced_eq_comap, ← map_le_iff_le_comap, filter.map_map]
+by rw [tendsto, tendsto, hg.induced, nhds_induced, ← map_le_iff_le_comap, filter.map_map]
 
 lemma embedding.continuous_iff {f : α → β} {g : β → γ} (hg : embedding g) :
   continuous f ↔ continuous (g ∘ f) :=
-by simp [continuous_iff_continuous_at, continuous_at, embedding.tendsto_nhds_iff hg]
+inducing.continuous_iff hg.1
 
 lemma embedding.continuous {f : α → β} (hf : embedding f) : continuous f :=
-hf.continuous_iff.mp continuous_id
+inducing.continuous hf.1
 
 lemma embedding.closure_eq_preimage_closure_image {e : α → β} (he : embedding e) (s : set α) :
   closure s = e ⁻¹' closure (e '' s) :=
-by ext x; rw [set.mem_preimage, ← closure_induced he.1, he.2]
+by { ext x,  rw [set.mem_preimage, ← closure_induced he.inj, he.induced] }
 
 end embedding
 
-
--- TODO: use embeddings from above!
-structure dense_embedding [topological_space α] [topological_space β] (e : α → β) : Prop :=
+structure dense_inducing [topological_space α] [topological_space β] (e : α → β)
+  extends inducing e : Prop  :=
 (dense   : ∀x, x ∈ closure (range e))
-(inj     : function.injective e)
-(induced : ∀a, comap e (nhds (e a)) = nhds a)
-
-theorem dense_embedding.mk'
-  [topological_space α] [topological_space β] (e : α → β)
-  (c     : continuous e)
-  (dense : ∀x, x ∈ closure (range e))
-  (inj   : function.injective e)
-  (H     : ∀ (a:α) s ∈ nhds a,
-    ∃t ∈ nhds (e a), ∀ b, e b ∈ t → b ∈ s) :
-  dense_embedding e :=
-⟨dense, inj, λ a, le_antisymm
-  (by simpa [le_def] using H a)
-  (tendsto_iff_comap.1 $ c.tendsto _)⟩
 
-namespace dense_embedding
+namespace dense_inducing
 variables [topological_space α] [topological_space β]
-variables {e : α → β} (de : dense_embedding e)
+variables {e : α → β} (di : dense_inducing e)
```

Comment:
I don't like this name `di` since it refers to `e`. Or you could rename `e` to `i` everywhere: this is probably more natural, but maybe too painful for what it's worth.

Reply:
Right, I'll switch to `i`. It's not fun to do, but we don't want historical accident to mess up notations.